lib/chef/cookbook/cookbook_version_loader.rb

```diff
@@ -78,6 +79,16 @@ def load
         # re-raise any exception that occurred when reading the metadata
         raise_metadata_error!
 
+        load_all_unignored_files
+
+        # TODO:
+        # 1. Can we safely pick out the segment files with fnmatch against the
+        #    file names in all_unignored_files (?)
+        # 2. If so, can we avoid running chefignore twice by running it against
+        #    all_unignored_files before segmenting?
+        #
+        # See: https://shane.io/2014/07/13/sobbing-with-ruby-file-globbing.html
```

Comment:
I seem to recall getting bitten by the differences between `Dir.glob` and `File.fnmatch` in the past. The blog post I linked says that the differences just come down to the two functions having different default option flags, which can be changed. But it still feels a bit risky. Thoughts?